### PR TITLE
simplify versions in getStackBottom

### PR DESCRIPTION
**Pull request by thewilsonator**
cc @Geod24

### CODE REVIEW

File: src/core/thread/osthread.d [hunk 1]
```diff
@@ -3332,15 +3332,17 @@ extern(C) void thread_processGCMarks( scope IsMarkedDg isMarked ) nothrow
...
-    version (CRuntime_Musl) int pthread_getattr_np(pthread_t, pthread_attr_t*);
-    version (CRuntime_UClibc) int pthread_getattr_np(pthread_t thread, pthread_attr_t* attr);
+    version (CRuntime_Glibc)  version = PThread_Getattr_NP;
+    version (CRuntime_Bionic) version = PThread_Getattr_NP;
+    version (CRuntime_Musl)   version = PThread_Getattr_NP;
+    version (CRuntime_UClibc) version = PThread_Getattr_NP;
+
+    version (FreeBSD)         version = PThread_Attr_Get_NP;
+    version (NetBSD)          version = PThread_Attr_Get_NP;
+    version (DragonFlyBSD)    version = PThread_Attr_Get_NP;
```
Comment (Geod24) [inline src/core/thread/osthread.d]: Should we have a `else static assert(0, "Unsupported platform")` ?

File: src/core/thread/osthread.d [hunk 2]
```diff
@@ -3332,15 +3332,17 @@ extern(C) void thread_processGCMarks( scope IsMarkedDg isMarked ) nothrow
 
 extern (C) @nogc nothrow
 {
-    version (CRuntime_Glibc) int pthread_getattr_np(pthread_t thread, pthread_attr_t* attr);
-    version (FreeBSD) int pthread_attr_get_np(pthread_t thread, pthread_attr_t* attr);
-    version (NetBSD) int pthread_attr_get_np(pthread_t thread, pthread_attr_t* attr);
-    version (OpenBSD) int pthread_stackseg_np(pthread_t thread, stack_t* sinfo);
-    version (DragonFlyBSD) int pthread_attr_get_np(pthread_t thread, pthread_attr_t* attr);
-    version (Solaris) int thr_stksegment(stack_t* stk);
```
Comment (Geod24) [inline src/core/thread/osthread.d]: Hum hum

File: src/core/thread/osthread.d [hunk 3]
```diff
@@ -3359,6 +3361,8 @@ package(core.thread) void* getStackTop() nothrow @nogc
 
 package(core.thread) void* getStackBottom() nothrow @nogc
 {
+
+
```
Comment (Geod24) [inline src/core/thread/osthread.d]: ?

**Comment by Geod24** [comment]
Note that *BSD have a module for `np` function. Maybe we should factor them out there, and just provide an alias ? https://github.com/dlang/druntime/search?utf8=%E2%9C%93&q=pthread_attr_get_np&type=

File: src/core/thread/osthread.d (hunk 1 above)
Comment (thewilsonator) [reply src/core/thread/osthread.d]: That is still handled in https://github.com/dlang/druntime/pull/2924/files#diff-f58c556db26f85763b2658851d550455L3487

**Comment by thewilsonator** [comment]
Probably, but I'm trying to not make any functional changes in this PR.

File: src/core/thread/osthread.d (hunk 2 above)
Comment (thewilsonator) [reply src/core/thread/osthread.d]: Fixed.

File: src/core/thread/osthread.d (hunk 3 above)
Comment (thewilsonator) [reply src/core/thread/osthread.d]: Fixed, that was a leftover from me discovering that you can assign versions inside a function.

File: src/core/thread/osthread.d (hunk 1 above)
Comment (thewilsonator) [reply src/core/thread/osthread.d]: Not here, as that would trigger on OSX and windows ;)

**Comment by thewilsonator** [comment]
This should be good to go.

**Pull request merged**